src/_pytest/nose.py

```diff
@@ -38,5 +41,6 @@ def call_optional(obj: object, name: str) -> bool:
         return False
     # If there are any problems allow the exception to raise rather than
     # silently ignoring it.
+    warnings.warn(NOSE_SUPPORT, stacklevel=2)
```

Comment:
not sure if we can unwind this into a user-space to be more helpful

Reply:
Indeed, but I think we should at least display the method name (and module), as well as the test that triggered this, if possible.

Reply:
any preferred way to handle this one @nicoddemus; we have the `item` in scope when called, could just pass the `nodeid` into a formattable warning? however some unit tests don't pass that obviously; I'm not sure how many edge cases there can be for this `nose` stuff;  I've everything else fixed up I think but as-is its something like:

```python
def call_optional(obj: object, name: str, nodeid: str) -> bool:
    method = getattr(obj, name, None)
    if method is None:
        return False
    is_fixture = getfixturemarker(method) is not None
    if is_fixture:
        return False
    if not callable(method):
        return False
    # If there are any problems allow the exception to raise rather than
    # silently ignoring it.
    warnings.warn(NOSE_SUPPORT.format(nodeid=nodeid, nose_method=name), stacklevel=2)
    method()
    return True
```
which I don't really like; ok to use `inspect` here to derive some of that info? or perhaps `__qualname__ / `__module__` would suffice on the existing function obj passed?